File: client/chunk.js
const http = require('http');
const fs = require('fs');

let file = fs.createWriteStream("a.ts");

let options = {
  host: "localhost",

  //query the file by key, this will do an internal redirect to assets location
  path: "/file/qazxswe",

  // THIS WILL GIVE A 404, since the route is internal
  //path: "/assets/a/interlaced/IPHONE5.ts",
  port: 80,
  headers: {
    //range: "bytes=0-1023"
    range: "bytes=0-9486856"
  }
}

http.get(options, (resp) => {
  console.log(resp.statusCode);
  console.log(resp.statusMessage);
  resp.pipe(file);
}).on("error", (e) => {
  console.error("Error", e);
})

File: client/chunk.py
import requests
import time
import sys

direct = sys.argv[1] == "direct"

if direct == True:
  url = "http://localhost/assets/ankit/interlaced/IPHONE5.ts"
else:
  url = "http://localhost/file/qazxswe"

chunk_size = 1024 * 1024
current_byte = 0

start = time.clock()

file_size = int(requests.head(url).headers['content-length'])
tmp_file = open("file.ts", "wb")
while current_byte <= file_size:
  #print 'Downloading %d - %d of %d' % (current_byte, current_byte + chunk_size, file_size)
  byte_range = "bytes=%d-%d" % (current_byte, current_byte + chunk_size)
  response = requests.get(url, headers={"Range": byte_range})
  for block in response.iter_content(chunk_size):
    tmp_file.write(block)

  current_byte = current_byte+chunk_size
tmp_file.close()

print time.clock() - start

#print "Done"
